### PR TITLE
Fix crashes on exit because of ResourceManager thread destruction

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8176/app_12691_PR.apk

File: libraries/networking/src/ResourceManager.cpp [hunk 1]
```diff
@@ -38,6 +38,11 @@ ResourceManager::ResourceManager(bool atpSupportEnabled) : _atpSupportEnabled(at
     _thread.start();
 }
 
+ResourceManager::~ResourceManager() {
+    _thread.exit();
```
[the remark this answers is not in the document]
Comment (birarda) [reply libraries/networking/src/ResourceManager.cpp]: FYI there's already a `cleanup` that some users of ResourceManager call manually that had the `quit()` and `wait()` on the thread

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8178/app_12691_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8178/HighFidelity-Beta-PR12691-00aa564d84f57e0add0b02bfde28183af663dbf4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8178/HighFidelity-Beta-PR12691-00aa564d84f57e0add0b02bfde28183af663dbf4.exe

**Pull request merged**